components/TARGET_PSA/spm/COMPONENT_SPM_MAILBOX/COMPONENT_NSPE/spm_client_proxy.c

```diff
@@ -0,0 +1,229 @@
+/* Copyright (c) 2018 ARM Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+// Includes
+// --------
+
+#include <stdio.h>
+#include "ipc_defs.h"
+#include "ipc_queue.h"
+#include "spm_client.h"
+#include "spm_messages.h"
+#include "mbed_assert.h"
+#include "cmsis_os2.h"
+
+
+// Globals
+// -------
+
+extern ipc_producer_queue_t *prod_queue;
+
+// API Functions Implementation
+// ----------------------------
+uint32_t psa_framework_version(void)
+{
+    return (uint32_t)PSA_FRAMEWORK_VERSION;
+}
+
+uint32_t psa_version(uint32_t sid)
+{
+    osRtxSemaphore_t res_sem_storage = {0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
We'll need to replace that with Mbed OS provided abstraction. I can accept it on condition that we will create a bug and fix it in 5.12. That's for all the places we use internal RTX types.